test/sql/cast/char.test

```diff
@@ -0,0 +1,87 @@
+# name: test/sql/cast/char.test
+# description: Issue #2588: Incorrect result with date conversion
+# group: [cast]
+
+statement ok
+PRAGMA enable_verification
+
+query III
+select 'asdf'::CHAR(3), 'asdf'::CHARACTER(3), 'asdf'::BPCHAR(3)
```

Comment:
Could we add some unicode tests, e.g.:

```sql
postgres=# select '🦆🦆🦆🦆'::char(3);
 bpchar 
--------
 🦆🦆🦆
```

Reply:
oh yeah that's a fun one.

Reply:
`LengthFun::IsCharacter` can maybe help here - it figures out if a byte is the first byte in a unicode codepoint.

Reply:
ended up using substring directly